static/app/components/profiling/flamegraphSearch.tsx

```diff
@@ -241,21 +242,26 @@ function FlamegraphSearch({
     }
   }, [search.results, search.index, onZoomIntoFrame]);
 
-  const handleChange: (value: string) => void = useCallback(
-    value => {
-      if (!value) {
-        dispatch({type: 'clear search'});
-        return;
-      }
-
-      dispatch({
-        type: 'set results',
-        payload: {
-          results: frameSearch(value, allFrames, searchIndex),
-          query: value,
-        },
-      });
-    },
+  const handleChange = useMemo(
+    () =>
+      debounce(value => {
```

Comment:
I have to be honest, I'm not a fan of debouncing because we end up penalizing the majority of the use cases because of occasional outliers. Do we know why search is slow here? Maybe we can do something there and optimize it, otherwise wdyt about enabling debouncing based on some heuristic like nb frames or something?

Reply:
>Every keystroke completely re-renders the page.

I think I noticed somthing similar already and it was due to us updating quert string on each keystroke and codebase that used useLocation

>Besides that, the obvious happens; we take the search results and have to re-render the canvas(s) on every keystroke, iterating over all frames at least twice, once for rects, once for text. etc. etc.

I think this is fine as long as we can do it in time.

One different way of tackling this issue without debouncing is to not allow searching on short queries. For example, if a user types something like a letter `a` into the input as their first char, it will likely match 90% of the frames and we'll end up drawing everything even though results are likely useless. If we for example limit shortest search to 3 char, we increase the specificity of the query at the expense of losing those first few characters which eod I'm not a fan of either.

But all of this is hard to say because we dont really know where the performance penalty comes from. For example, if searching through something like 10k frames is slow, then debouncing might make sense, but maybe has more to do with how we end up rerendering the entire page or very likely a combo of both. In either way, I'd look at improving that part before settling on debouncing or make it conditional for some N of frames.